src/vs/editor/contrib/linesOperations/linesOperations.ts

```diff
@@ -316,13 +316,13 @@ class DeleteLinesAction extends EditorAction {
 			return a.startLineNumber - b.startLineNumber;
 		});
 
-		// Merge delete operations on consecutive lines
+		// Merge delete operations which are adjacent or overlapping
 		let mergedOperations: IDeleteLinesOperation[] = [];
 		let previousOperation = operations[0];
 		for (let i = 1; i < operations.length; i++) {
-			if (previousOperation.endLineNumber + 1 === operations[i].startLineNumber) {
+			if (previousOperation.endLineNumber + 1 >= operations[i].startLineNumber) {
 				// Merge current operations into the previous one
-				previousOperation.endLineNumber = operations[i].endLineNumber;
+				previousOperation.endLineNumber = Math.max(previousOperation.endLineNumber, operations[i].endLineNumber);
```

Comment:
Math.max is needed in case the first operation is a subset of the second operation. This can actually happen I checked.